File: nhglib-core/src/io/github/voidzombie/nhglib/data/models/serialization/physics/shapes/SphereShapeJson.java
package io.github.voidzombie.nhglib.data.models.serialization.physics.shapes;

import com.badlogic.gdx.physics.bullet.collision.btSphereShape;
import com.badlogic.gdx.utils.JsonValue;

/**
 * Created by Fausto Napoli on 16/05/2017.
 */
public class SphereShapeJson extends ShapeJson {
    private btSphereShape shape;

    @Override
    public void parse(JsonValue jsonValue) {
        float radius = jsonValue.getFloat("radius");
        shape = new btSphereShape(radius);
    }

    @Override
    public btSphereShape get() {
        return shape;
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/data/models/serialization/physics/shapes/SphereShapeJson.kt
package io.github.movementspeed.nhglib.data.models.serialization.physics.shapes;

import com.badlogic.gdx.utils.JsonValue;
import io.github.movementspeed.nhglib.physics.models.SphereRigidBodyShape;

/**
 * Created by Fausto Napoli on 16/05/2017.
 */
public class SphereShapeJson extends ShapeJson {
    private SphereRigidBodyShape shape;

    @Override
    public void parse(JsonValue jsonValue) {
        float radius = jsonValue.getFloat("radius");
        shape = new SphereRigidBodyShape(radius);
    }

    @Override
    public SphereRigidBodyShape get() {
        return shape;
    }
}
